pallets/thea-executor/src/lib.rs

```diff
@@ -659,50 +662,68 @@ pub mod pallet {
 			// Get the metadata
 			let metadata =
 				<Metadata<T>>::get(deposit.asset_id).ok_or(Error::<T>::AssetNotRegistered)?;
-			let deposit_amount = deposit.amount_in_native_decimals(metadata); // Convert the decimals configured in metadata
 
-			if !frame_system::Pallet::<T>::account_exists(&deposit.recipient) {
-				let path = sp_std::vec![
-					polkadex_primitives::AssetId::Asset(deposit.asset_id),
-					polkadex_primitives::AssetId::Polkadex
-				];
-				let amount_out: T::AssetBalanceAdapter = T::ExistentialDeposit::get().into();
-				Self::resolve_mint(&Self::thea_account(), deposit.asset_id.into(), deposit_amount)?;
+			let deposit_amount = deposit.amount_in_native_decimals(metadata); // Convert the decimals configured in metadata
 
-				// If swap doesn't work then it will in the system account - thea_account()
-				if let Ok(fee_amount) = T::Swap::swap_tokens_for_exact_tokens(
-					Self::thea_account(),
-					path,
-					amount_out.into(),
-					Some(deposit_amount),
-					deposit.recipient.clone(),
-					true,
-				) {
+			let final_deposit_amount =
+				if !frame_system::Pallet::<T>::account_exists(&deposit.recipient)
+					&& deposit.asset_id != AssetId::Polkadex
+				{
+					let path =
+						sp_std::vec![deposit.asset_id, polkadex_primitives::AssetId::Polkadex];
+					let amount_out: T::AssetBalanceAdapter = T::ExistentialDeposit::get().into();
+					Self::resolve_mint(
+						&Self::thea_account(),
+						deposit.asset_id.into(),
+						deposit_amount,
+					)?;
+					// If swap doesn't work then it will in the system account - thea_account()
+					let fee_amount = T::Swap::swap_tokens_for_exact_tokens(
+						Self::thea_account(),
+						path,
+						amount_out.into(),
+						Some(deposit_amount),
+						deposit.recipient.clone(),
+						true,
+					)?;
+					let final_amount = deposit_amount.saturating_sub(fee_amount);
 					Self::resolve_transfer(
 						deposit.asset_id.into(),
 						&Self::thea_account(),
 						&deposit.recipient,
-						deposit_amount.saturating_sub(fee_amount),
+						final_amount,
 					)?;
-				}
-			} else {
-				Self::resolver_deposit(
-					deposit.asset_id.into(),
-					deposit_amount,
-					&deposit.recipient,
-					Self::thea_account(),
-					1u128,
-					Self::thea_account(),
-				)?;
-			}
+					final_amount
+				} else {
+					Self::resolver_deposit(
+						deposit.asset_id.into(),
+						deposit_amount,
+						&deposit.recipient,
+						Self::thea_account(),
+						1u128,
+						Self::thea_account(),
+					)?;
+					deposit_amount
+				};
 
 			// Emit event
 			Self::deposit_event(Event::<T>::DepositClaimed(
 				deposit.recipient.clone(),
 				deposit.asset_id,
-				deposit.amount_in_native_decimals(metadata),
+				final_deposit_amount,
 				deposit.id,
 			));
+
+			match deposit.extra {
```

Comment:
how is parachain mapping parsing xcm to direct deposit and normal deposits?

Reply:
@zktony  will be updating that, but in short, it will be having a X2 junction in multilocation.

Reply:
Implemented